auto_gptq/modeling/_base.py

```diff
@@ -1112,16 +1112,17 @@ def skip(*args, **kwargs):
 
                     model = convert_to_marlin(model, quant_linear_class, quantize_config, repack=True)
 
+                    weights = {k: v.clone().contiguous() for k, v in model.state_dict().items()}
```

Comment:
Why is this needed? This would double memory.

Reply:
I had this [problem](https://github.com/huggingface/safetensors/issues/202) and used it.
If there is another better method, please apply it.

Reply:
Does removing the `clone()` work?

Reply:
> Does this still work when removing the `clone()`?

No. Not work.

Reply:
Thanks, could you share a repro of the issue?

Reply:
```python
from transformers import AutoTokenizer, TextGenerationPipeline
from auto_gptq import AutoGPTQForCausalLM, BaseQuantizeConfig
import logging

logging.basicConfig(
    format="%(asctime)s %(levelname)s [%(name)s] %(message)s", level=logging.INFO, datefmt="%Y-%m-%d %H:%M:%S"
)

pretrained_model_dir = "facebook/opt-125m"
quantized_model_dir = "opt-125m-4bit"

tokenizer = AutoTokenizer.from_pretrained(pretrained_model_dir, use_fast=True)
examples = [
    tokenizer(
        "auto-gptq is an easy-to-use model quantization library with user-friendly apis, based on GPTQ algorithm."
    )
]

quantize_config = BaseQuantizeConfig(
    bits=4,  # quantize model to 4-bit
    group_size=128,  # it is recommended to set the value to 128
    sym=True,
    desc_act=False,  # set to False can significantly speed up inference but the perplexity may slightly bad
)

# load un-quantized model, by default, the model will always be loaded into CPU memory
model = AutoGPTQForCausalLM.from_pretrained(pretrained_model_dir, quantize_config)

# quantize model, the examples should be list of dict whose keys can only be "input_ids" and "attention_mask"
model.quantize(examples)

# save quantized model
model.save_quantized(quantized_model_dir)


# load quantized model to the first GPU
model = AutoGPTQForCausalLM.from_quantized(quantized_model_dir, device="cuda:0", use_marlin=True)

# download quantized model from Hugging Face Hub and load to the first GPU
# model = AutoGPTQForCausalLM.from_quantized(repo_id, device="cuda:0", use_safetensors=True, use_triton=False)

# inference with model.generate
print(tokenizer.decode(model.generate(**tokenizer("auto_gptq is", return_tensors="pt").to(model.device))[0]))
```

Reply:
Thank you, this is an asymmetry in safetensors vs pytorch serialization, whereh safetensors is unable to handle shared weights https://github.com/huggingface/safetensors/issues/202.

I fixed it in https://github.com/AutoGPTQ/AutoGPTQ/pull/536.